### ISSUE TITLE
Secure Connection Failed with Squid and SSL

### ISSUE THREAD

**Issue by hardhub** [foxyproxy/firefox-extension#32]
It says PR_CONNECT_RESET_ERROR

But if set to use proxy for all urls ignoring patterns then all works!

**Comment by ericjung** [comment]
By secure connection, do you mean an SSL proxy server? Or just a website with SSL?

**Comment by hardhub** [comment]
> By secure connection, do you mean an SSL proxy server? Or just a website with SSL?

Both... squid config looks like this (squid 3.5):

```
https_port 8888 cert=my.crt key=my.key \
    cipher=ECDHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-SHA384:AES256-SHA \
    version=4 # TLS1 Only \
    options=NO_SSLv2 NO_SSLv3
```

**Comment by ericjung** [comment]
Neither me nor my colleague can reproduce this with an SSL squid proxy server. Is it possible for you to share access to it temporarily? You can email me directly, of course. eric AT ericjung.net

**Comment by johnnybubonic** [comment]
@hardhub try a couple things for me:

**Test case 1**
* Disable: _cipher_, _version_, _options_

**Test case 2**
* Change: _options=NO_SSLv2 NO_SSLv3_ to _options=NO_SSLv2,NO_SSLv3_

**Test case 3**
Create a brand new [Firefox profile](https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Multiple_profiles) and test using your regular settings and recent release FoxyProxy.

I'm especially interested in the results of case 3, as I *suspect* this is the case of a corrupt security store/cache, but please let us know the results of all three.

**Comment by hardhub** [comment]
I have tried your 3rd case (by 2 different ways) and it works.

- I used Developer Edition and exported/imported settings from main Firefox.
- I removed plugin and installed it back, then imported old settings, so now it works in primary browser as well.

So suggestion looks like correct. So there was no a need to test 1 and 2.  
But before that I tested it with version version = 1 (auto).  
No results of course...

But removing and adding plugin back makes sense to workaround this issue.
